test/testdrive/kafka-json-sinks.td

```diff
@@ -0,0 +1,110 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+> CREATE VIEW unnamed_cols AS SELECT 1, 2 AS b, 3;
+
+> CREATE SINK unnamed_cols_sink FROM unnamed_cols
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'unnamed-cols-sink'
+  FORMAT JSON
+
+$ kafka-verify format=json sink=materialize.public.unnamed_cols_sink key=false
+{"before": null, "after": {"row": {"column1": 1, "b": 2, "column3": 3}}}
+
+> CREATE SINK unnamed_cols_upsert FROM unnamed_cols
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'unnamed-upsert'
+  KEY (b)
+  FORMAT JSON
+  ENVELOPE UPSERT
+
+$ kafka-verify format=json sink=materialize.public.unnamed_cols_upsert key=true
```

Comment:
Ahhh, `kafka-ingest` determines `key` by checking if there is a registered key schema. For JSON we don't need it because we can check without the schema. It's messy :sweat_smile:

Reply:
It is messy! I agree that it's not great to add asymmetric parameters here, I'll think about it more